src/extractors/positiveFlux.ts

```diff
@@ -1,26 +1,25 @@
 import { normalizeToOne } from "../utilities";
 
 export default function ({
-  complexSpectrum,
-  previousComplexSpectrum,
+  ampSpectrum,
+  previousAmpSpectrum,
 }: {
-  complexSpectrum: { real: number[]; imag: number[] };
-  previousComplexSpectrum: { real: number[]; imag: number[] };
+  ampSpectrum: Float32Array;
+  previousAmpSpectrum: Float32Array;
 }): number {
-  if (!previousComplexSpectrum) {
+  if (!previousAmpSpectrum) {
     return 0;
   }
 
-  const normalizedRealComponent = normalizeToOne(complexSpectrum.real);
-  const previousNormalizedRealComponent = normalizeToOne(
-    previousComplexSpectrum.real
-  );
+  const normalizedMagnitudeSpectrum = normalizeToOne(ampSpectrum);
+  const previousNormalizedMagnitudeSpectrum =
+    normalizeToOne(previousAmpSpectrum);
```

Comment:
I wonder if this is correct. Should they be normalised between themselves? Should they be normalised at all? I am not sure normalising each one to 1 separately makes sense. Can you share the source this was taken from?

FWIW, Matlab [implements it](https://www.mathworks.com/help/audio/ref/spectralflux.html#mw_ce89ba6d-8ba7-4ae4-aa16-7d329014ec1b) without normalisation, it seems.

Reply:
I'm not sure - I added normalization because [wikipedia](https://en.wikipedia.org/wiki/Spectral_flux#Variations) implies that only _some_ implementations do not normalize. I think it's useful because it makes this feature purely about the spectral content, and resilient to overall volume changes, which can be measured much better with rms and loudness. But, I haven't gone looking for other implementations yet.